device/mellanox/x86_64-mlnx_lssn2700-r0/plugins/sfputil.py

```diff
@@ -64,6 +92,22 @@ def get_presence(self, port_num):
             return True
 
         return False
+        
+    def get_presence_status_file_path(self, port_num):
```

Comment:
Most of the code added in this PR is duplicated among all the plugins. Is there a reason this functionality can't live in sfputilbase.py? Will it not be shared by other platforms?

Reply:
the file path is specific for each platform, it's even various among platforms from one vendor, so this has to be specific for each platform.